app/src/keymap.c

```diff
@@ -74,6 +74,12 @@ static struct zmk_behavior_binding zmk_sensor_keymap[ZMK_KEYMAP_LAYERS_LEN]
 
 #endif /* ZMK_KEYMAP_HAS_SENSORS */
 
+static void raise_layer_event(zmk_keymap_layers_state_t prior_state) {
+    LOG_DBG("layer_changed: layer %d, old state %x, state %x", zmk_keymap_highest_layer_active(),
+            prior_state, zmk_keymap_layer_state());
+    ZMK_EVENT_RAISE(create_layer_state_changed(prior_state));
```

Comment:
The event should only be raised if the prior state is not equal to the current state.

Reply:
In my use case I'd like to respond to a layer change event _even if the layer does not change_.  Consumers of the event that only want layer change events where the layer changes can filter for that using the state included in the event.

Reply:
Then "later state changed" is a hood wink...

This should be renamed to "layer state event" as it could be *no* change...

Reply:
That makes more sense.  I'll do that.

Reply:
I'm curious what your usecase is for getting layer events even if the layer doesn't change. Can you help me understand?

Reply:
Hmm, now that I looked at the code, all other events follow a naming convention of zmk_XXX_YYY_changed.  Changing it to "layer state event" would depart from that convention.  I think documenting the zmk_layer_state_changed and following the naming convention would be a better approach.

Reply:
If I'm reading the BLE profile code correctly, it'll raise a zmk_ble_active_profile_changed even if it's the same profile as before - so we have precedent for the current behavior and naming convention.

Reply:
So I suggest leaving as-is.

Reply:
For the previous incarnation of the layer state changed, we decided to not send events for layers that stay the same, so there is precendent for both. Can you explain what the reason is you need this, because I genuine would like to understand why you'd want this.

Reply:
You're correct I realized I could test it in my configuration and it does not work.  I'll need to see about changing that behavior - I would like to get the event.

I setting leds in response to users changing layers.  I'd like to flash the leds even if they select the same layer so they have feedback.  I'd like to do the same with the BLE profiles too.

Reply:
Hm yeah sounds reasonable. I don't see another way to do this than to send the 'layer state changed' event unconditionally. :+1:

Reply:
So we keep the current name _zmk_layer_state_changed_ to match the existing convention?

Reply:
In the current code the zmk_battery_state_changed event is sent without a change so I think it's reasonable to leave the name as-it.